test/errors.ts

```diff
@@ -0,0 +1,97 @@
+import { InstrumentationEventName } from "../src/instrumentation";
+import { activateCore, cleanupFixture, getAllSentEvents, setupFixture, triggerSend } from "./utils";
+
+import * as chai from "chai";
+import * as errors from "../src/errors";
+
+let assert = chai.assert;
+
+describe("Error Tests", () => {
+    beforeEach(setupFixture);
+    afterEach(cleanupFixture);
+
+    it("checks that a single error event is logged", (done) => {
+        let syntheticEvent = document.createEvent("CustomEvent");
+        let message = "sample error text";
+        let filename = "sample filename";
+        let lineno = "sample error text";
+        let colno = "sample error text";
+        syntheticEvent.initEvent("error", true, true);
+        syntheticEvent["message"] = message;
+        syntheticEvent["filename"] = filename;
+        syntheticEvent["lineno"] = lineno;
+        syntheticEvent["colno"] = colno;
+        errors.logError(syntheticEvent);
```

Comment:
Should you have the window dispatch this event, rather than invoking a function directly ?